fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java

```diff
@@ -1695,16 +1700,201 @@ private void handleShowRestore() throws AnalysisException {
+
+                                    List<String> privList = new ArrayList<>();
+                                    for (Map.Entry<String, Action> actionEntry
+                                            : ObjectType.CATALOG.getActionMap().entrySet()) {
+                                        if (privilegeEntry.getActionSet().contains(actionEntry.getValue())) {
+                                            privList.add(actionEntry.getValue().getName());
+                                        }
+                                    }
+                                    GrantPrivilegeStmt grantPrivilegeStmt =
+                                            new GrantPrivilegeStmt(privList, ObjectType.CATALOG.name(), userIdentity);
+                                    info.add(AstToStringBuilder.toString(grantPrivilegeStmt));
+                                }
+                            } else {
+                                info.add("default");
+                            }
+
+                            GrantPrivilegeStmt grantPrivilegeStmt =
+                                    new GrantPrivilegeStmt(new ArrayList<>(), objectType.name(), userIdentity);
+                            grantPrivilegeStmt.setObjectType(objectType);
+                            grantPrivilegeStmt.setActionList(privilegeEntry.getActionSet());
+                            grantPrivilegeStmt.setObjectList(Lists.newArrayList(privilegeEntry.getObject()));
+                            info.add(AstToStringBuilder.toString(grantPrivilegeStmt));
+
+                            infos.add(info);
+                        }
+                    }
+
+                    resultSet = new ShowResultSet(showStmt.getMetaData(), infos);
+                }
+            } catch (PrivilegeException e) {
+                throw new SemanticException(e.getMessage());
+            }
+        } else {
+            List<List<String>> infos = GlobalStateMgr.getCurrentState().getAuth().getGrantsSQLs(showStmt.getUserIdent());
+            resultSet = new ShowResultSet(showStmt.getMetaData(), infos);
+        }
     }
 
-    private void handleShowRoles() {
+    private void handleShowRoles() throws AnalysisException {
         ShowRolesStmt showStmt = (ShowRolesStmt) stmt;
-        // TODO(yiming): handle show roles in new RBAC privilege framework
-        List<List<String>> infos = GlobalStateMgr.getCurrentState().getAuth().getRoleInfo();
-        resultSet = new ShowResultSet(showStmt.getMetaData(), infos);
+
+        if (GlobalStateMgr.getCurrentState().isUsingNewPrivilege()) {
+            try {
+                List<List<String>> infos = new ArrayList<>();
+                PrivilegeManager privilegeManager = GlobalStateMgr.getCurrentState().getPrivilegeManager();
+                Map<Long, RolePrivilegeCollection> rolePrivilegeCollectionMap = privilegeManager.getRoleIdToPrivilegeCollection();
+                for (Map.Entry<Long, RolePrivilegeCollection> rolePrivilegeCollectionEntry
```

Comment:
we need the role lock protection here, but currently the lock/unlock method of roleLock is private, we better put the loop logic in a method in PrivilegeManger to avoid violating encapsulation.